File: PGV/unidad5/mi-tasks-api/src/main/java/com/docencia/tasks/adapters/mapper/TaskMapper.java
package com.docencia.tasks.adapters.mapper;

import java.util.List;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import com.docencia.tasks.adapters.in.api.TaskRequest;
import com.docencia.tasks.adapters.in.api.TaskResponse;
import com.docencia.tasks.domain.model.Task;
import com.docencia.tasks.entitys.TaskEntity;

@Mapper(componentModel = "spring")
public interface TaskMapper {

    @Mapping(source = "identificador", target = "id")
    @Mapping(source = "titulo", target = "title")
    @Mapping(source = "descripcion", target = "description")
    @Mapping(source = "completado", target = "completed")
    public Task requestToTask(TaskRequest source);

    public List<Task> toTaskList(List<TaskRequest> requests);

    public Task entityToTask(TaskEntity source);

    public List<Task> entityToTaskList(List<TaskEntity> entities);

    public TaskResponse taskToResponse(Task task);
}

File: AED/unidad4/Tareas_angular/tarea2_Fuera de plazo 3.4/gestor-tareas/src/app/services/task-mapper.ts
import { Task, NewTask } from '../models/task.model';

export interface ApiTask {
  id: number;
  title: string;
  description?: string;
  completed: boolean;
}

export type ApiNewTask = Omit<ApiTask, 'id'>;

export function fromApiTask(a: ApiTask): Task {
  return {
    id: a.id,
    titulo: a.title,
    descripcion: a.description,
    completada: a.completed,
  };
}

export function toApiNewTask(t: NewTask): ApiNewTask {
  return {
    title: t.titulo,
    description: t.descripcion ?? '',
    completed: t.completada,
  };
}